fal/controllers/bench_swaps.py

```diff
@@ -0,0 +1,103 @@
+from __future__ import annotations
+
+from typing import List, Tuple, TYPE_CHECKING
+import io
+import http.client
+import re
+import gzip
+import configparser
+
+from fal.orm import Secret, mfalncfm_main
+from fal.models import Team, Season, Anime
+
+if TYPE_CHECKING:
+    from sqlalchemy.orm import Session
+
+httpconn = http.client.HTTPSConnection("myanimelist.net")
+
+config = configparser.ConfigParser()
+config.read("config.ini")
+
+active_len = config.getint("season info", "num-active-on-team")
+
+PARAMS = ""
+
+
+def get_swaps(
+    session: Session,
+    page: int = config.getint("weekly info", "bench-swap-page-start"),
+    actions: List[Tuple] = [],
+) -> List[Tuple]:
+    """
+    NOTE: copied from old FAL script and converted to Python 3 and uses our modern stack
+
+    Collects swaps for the current week.
+
+    This function accesses the swap thread.
+
+    @param page page identification for threads
+    @param actions list of tuples (username, post content)
+
+    @return list of tuples (username, post content = swap)
+    """
+    header = dict(config.items("myanimelist.net request header"))
+
+    with mfalncfm_main.session_scope() as session:
+        secret_request_headers = (
+            session.query(Secret)
+            .filter(Secret.context == "myanimelist.net request header")
+            .all()
+        )
+        for row in secret_request_headers:
+            header[row.key] = row.value
+
+    httpconn.connect()
+    httpconn.request(
+        "GET",
+        "/forum/?topicid=%s&show=%i"
+        % (config.getint("myanimelist", "bench-swap-thread-id"), (page - 1) * 50),
+        PARAMS,
+        header,
+    )
+    res = httpconn.getresponse()
+    if res.getheader("Content-Encoding") == "gzip":
+        response = gzip.GzipFile(fileobj=io.BytesIO(res.read())).read()
+    else:
+        response = res.read()
+    httpconn.close()
+    pattern = 'postnum">(\d+).+?row2.+?<strong>(.+?)</strong>.+?row1.+?message\d.+?class="clearfix">(.+?)</div>'
+    actions.extend(re.findall(pattern, str(response), re.S))
```

Comment:
I think we should try to use [Beautiful Soup](https://beautiful-soup-4.readthedocs.io/en/latest/) to web scrape instead of using regex. We can tackle it in a future PR as it's not urgent.

Reply:
Agreed